module/zfs/vdev.c

```diff
@@ -1820,6 +1820,21 @@ vdev_dtl_empty(vdev_t *vd, vdev_dtl_type_t t)
 }
 
 /*
+ * Returns B_TRUE if vdev determines offset needs to be resilvered.
+ */
+boolean_t
+vdev_dtl_need_resilver(vdev_t *vd, uint64_t offset, size_t psize)
+{
+	ASSERT(vd != vd->vdev_spa->spa_root_vdev);
```

Comment:
I don't understand this assertion. Shouldn't it be ASSERT3P(vd, ==, vd->vdev_top) instead? It's checking a DVA, so the function only makes sense for a top-level vdev.